In [5]:
19*48*48*48 # 2101248 ~ 2M data points for each complex

2101248

In [6]:
48*48*48 # 110592 ~ 100K data points for each feature

110592

# HDF Files

In [7]:
import h5py

filename = "data/pdbbind2016_core_test.hdf"


with h5py.File(filename, 'r') as f:
    protein_ids = list(f.keys())

    print(f"Number of Proteins: {len(protein_ids)}")
    print(f"Sample Proteins: {protein_ids[:5]}")

Number of Proteins: 285
Sample Proteins: ['1a30', '1bcu', '1bzc', '1c5z', '1e66']


In [8]:
import h5py

def print_structure(name, obj):
    print(name, type(obj))

filename = "data/pdbbind2016_core_test.hdf"

with h5py.File(filename, 'r') as f:
    f.visititems(print_structure)

1a30 <class 'h5py._hl.group.Group'>
1a30/pybel <class 'h5py._hl.group.Group'>
1a30/pybel/processed <class 'h5py._hl.group.Group'>
1a30/pybel/processed/pdbbind <class 'h5py._hl.group.Group'>
1a30/pybel/processed/pdbbind/data <class 'h5py._hl.dataset.Dataset'>
1bcu <class 'h5py._hl.group.Group'>
1bcu/pybel <class 'h5py._hl.group.Group'>
1bcu/pybel/processed <class 'h5py._hl.group.Group'>
1bcu/pybel/processed/pdbbind <class 'h5py._hl.group.Group'>
1bcu/pybel/processed/pdbbind/data <class 'h5py._hl.dataset.Dataset'>
1bzc <class 'h5py._hl.group.Group'>
1bzc/pybel <class 'h5py._hl.group.Group'>
1bzc/pybel/processed <class 'h5py._hl.group.Group'>
1bzc/pybel/processed/pdbbind <class 'h5py._hl.group.Group'>
1bzc/pybel/processed/pdbbind/data <class 'h5py._hl.dataset.Dataset'>
1c5z <class 'h5py._hl.group.Group'>
1c5z/pybel <class 'h5py._hl.group.Group'>
1c5z/pybel/processed <class 'h5py._hl.group.Group'>
1c5z/pybel/processed/pdbbind <class 'h5py._hl.group.Group'>
1c5z/pybel/processed/pdbbind/data

In [9]:
import h5py

import numpy as np
import pandas as pd

filename = "data/pdbbind2016_core_test.hdf"

with h5py.File(filename, "r") as pdb:
   
    # understand 1bzc data
    complex = "1bzc"



    # If a_group_key is a group name, 
    # this gets the object names in the group and returns as a list
    data5 = list(pdb[complex]['pybel']['processed']['pdbbind']['data'])


    # If a_group_key is a dataset name, 
    # this gets the dataset values and returns as a list
    # data = list(f[a_group_key])
    print("d5", type(data5), len(data5))

    data5_np = np.array(data5)
    print("d5-np", type(data5), data5_np.shape,len(data5_np[1]), data5_np[1], )




d5 <class 'list'> 396
d5-np <class 'list'> (396, 22) 22 [-1.0471725  2.7058601 -5.6898966  0.         0.         0.
  1.         0.         0.         0.         0.         0.
  2.         1.         1.        -0.5666     1.         0.
  1.         0.         0.         0.       ]


### HDF data explained

- 285 complexes
- For each complex, hiearchy is "pybel","processed", "pdbbind", "data"
- data.shape is (396, 22) for "1bzc" protein so possibly they have a way to reduce it to 392 rows, because actual process is ~2M(19*48*48*48) for each complex and ~100K(48*48*48) data points for each feature in complex.
- First 3 attributes are xyz co-ord, rest are features


# .npy files

In [10]:
import numpy as np
refined_train_fc6 = np.load('data/refined_train_fc6.npy')
refined_train_fc6.shape

(4779, 6)

In [11]:
import numpy as np
refined_train_fc10 = np.load('data/refined_train_fc10.npy')
refined_train_fc10.shape

(4779, 10)

In [12]:
import numpy as np
refined_val_fc6 = np.load('data/refined_val_fc6.npy')
refined_val_fc6.shape

(537, 6)

In [13]:
import numpy as np
refined_val_fc10 = np.load('data/refined_val_fc10.npy')
refined_val_fc10.shape

(537, 10)

In [14]:
import numpy as np
core_test_fc6 = np.load('data/core_test_fc6.npy')
core_test_fc6.shape

(285, 6)

In [15]:
import numpy as np
core_test_fc10 = np.load('data/core_test_fc10.npy')
core_test_fc10.shape

(285, 10)